Src/AutoNSubstitute/AutoNSubstitute.csproj

```diff
@@ -64,6 +64,7 @@
   <ItemGroup>
     <Compile Include="AutoConfiguredNSubstituteCustomization.cs" />
     <Compile Include="AutoNSubstituteCustomization.cs" />
+    <Compile Include="AutoSubstituteCustomization.cs" />

```

Comment:
I will update the existing customization classes and remove the `AutoSubstituteCustomization` class then.

You've raised a good point - the existing `NSubstituteBuilder` and the new `SubstituteBuilder` are indeed different; the first handles abstract `Type` requests while the second handles the new `SubstituteRequest`. I agree that replacing the existing type with the new would be a breaking change. The new `SubstituteRelay`, however, is a much closer match because it also handles abstract `Type` requests, so if you prefer to deprecate the existing `NSubstituteBuilder` this class could be a replacement.

I think we still need to agree on the naming convention for the new classes. I don't understand why renaming the new `SubstituteXxx` classes to include the `N` prefix would be a breaking change. Aside from the new `SubstituteBuilder` there are no naming collisions with the existing classes and making the new class use a different, more descriptive name, like `NSubstituteRequestHandler` would solve this particular problem.